tests/unit/test_backup.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Synapse backup unit tests."""
+
+from secrets import token_hex
+from unittest.mock import MagicMock
+
+import boto3
+import pytest
+from botocore.exceptions import BotoCoreError, ClientError
+
+import backup
+
+
+def test_s3_relation_validation_fails_when_region_and_endpoint_not_set():
```

Comment:
Are there any other validations we should check?

Reply:
Yes, the correct case. And a hidden bug was in them...